### PR TITLE
CONSOLE-3683: Convert FireMan component in list-page.tsx from class c…

### PR DESCRIPTION
**Pull request by tvu20**
…omponent to functional component

FireMan component in list-page.tsx converted from a class component to a functional component in order to use react-router v6 hooks.

### CODE REVIEW

File: frontend/public/components/factory/list-page.tsx [hunk 1]
```diff
@@ -185,184 +185,167 @@ export type FireManProps = {
...
-      }
-      const params = new URLSearchParams(window.location.search);
-      if (options) {
-        params.set(filterName, options);
-      } else {
-        params.delete(filterName);
-      }
-      const url = new URL(window.location.href);
-      history.replace(`${url.pathname}?${params.toString()}${url.hash}`);
+  const updateURL = (filterName, options) => {
```
Comment (jhadvig) [inline frontend/public/components/factory/list-page.tsx:239]: lets type these params

File: frontend/public/components/factory/list-page.tsx [hunk 2]
```diff
@@ -185,184 +185,167 @@ export type FireManProps = {
...
-      // TODO: (ggreer) lame blacklist of query args. Use a whitelist based on resource filters
-      if (['q', 'kind', 'orderBy', 'sortBy'].includes(filterName)) {
-        return;
-      }
-      if (filterName.indexOf(storagePrefix) === 0) {
-        return;
-      }
-      this.state.reduxIDs.forEach((id) => this.props.filterList(id, filterName, options));
-      this.updateURL(filterName, options);
+  const applyFilter = (filterName, options) => {
```
Comment (jhadvig) [inline frontend/public/components/factory/list-page.tsx:254]: lets type these params here as well

File: frontend/public/components/factory/list-page.tsx [hunk 3]
```diff
@@ -185,184 +185,167 @@ export type FireManProps = {
...
+      return;
     }
+    reduxIDs.forEach((id) => props.filterList(id, filterName, options));
+    updateURL(filterName, options);
+  };
 
-    UNSAFE_componentWillMount() {
-      const params = new URLSearchParams(window.location.search);
-      params.forEach((v, k) => this.applyFilter(k, v));
+  const runOrNavigate = (itemName) => {
```
Comment (jhadvig) [inline frontend/public/components/factory/list-page.tsx:266]: same

File: frontend/public/components/factory/list-page.tsx [hunk 4]
```diff
@@ -185,184 +185,167 @@ export type FireManProps = {
...
+  React.useEffect(() => {
+    const reduxid = resources.map((r) =>
+      makeReduxID(
+        kindObj(r.kind),
+        makeQuery(r.namespace, r.selector, r.fieldSelector, r.name),
+        cluster, // TODO remove multicluster
+      ),
+    );
+
+    if (_.isEqual(reduxid, reduxIDs)) {
```
Comment (jhadvig) [inline frontend/public/components/factory/list-page.tsx:230]: nit:
```suggestion
    if (_.isEqual(reduxID, reduxIDs)) {
```

File: frontend/public/components/factory/list-page.tsx [hunk 5]
```diff
@@ -185,184 +185,167 @@ export type FireManProps = {
...
+    params.forEach((v, k) => applyFilter(k, v));
+
+    const reduxid = resources.map((r) =>
+      makeReduxID(
+        kindObj(r.kind),
+        makeQuery(r.namespace, r.selector, r.fieldSelector, r.name),
+        cluster, // TODO remove multicluster
+      ),
+    );
+    setReduxIDs(reduxid);
```
Comment (jhadvig) [inline frontend/public/components/factory/list-page.tsx:217]: nit:
```suggestion
    setReduxIDs(reduxID);
```

File: frontend/public/components/factory/list-page.tsx [hunk 6]
```diff
@@ -185,184 +185,167 @@ export type FireManProps = {
   cluster?: string; // TODO remove multicluster
 };
 
-type FireManState = {
-  reduxIDs: string[];
-  expand?: boolean;
-};
+export const FireMan = (props) => {
```
Comment (jhadvig) [inline frontend/public/components/factory/list-page.tsx:188]: Lets type the props with `FireManProps`

**Review by jhadvig — COMMENTED** [review]
Thanks @tvu20
PR looks good, adding couple of small commets. PTAL :)

**Review by jhadvig — APPROVED** [review]
/lgtm
/approve

**Comment by XiyunZhao** [comment]
This PR has been tested on private build, with manual and regression testing, and no issue was found
/label qe-approved

**Pull request merged**